contracts/SlashIndicator.sol

```diff
@@ -158,6 +188,73 @@ contract SlashIndicator is ISlashIndicator,System,IParamSubscriber, IApplication
     emit indicatorCleaned();
   }
 
+  function handleFinalityReport(FinalityEvidence calldata _evidence) external onlyInit oncePerBlock noReentrant {
+    require(
+      _evidence.numA + finalityDistance >= _evidence.numB || _evidence.numB + finalityDistance >= _evidence.numA,
+      'too long distance between blocks'
+    );
+    require(
+      _evidence.numA <= block.number &&
+      _evidence.numA + 256 >= block.number &&
+      _evidence.numB <= block.number &&
+      _evidence.numB + 256 >= block.number,
+      'block number out of range'
+    );
+
+    require(IBSCValidatorSet(VALIDATOR_CONTRACT_ADDR).isCurrentValidator(_evidence.valAddr), "not current validator");
+
+    (address[] memory vals, bytes[] memory voteAddrs) = IBSCValidatorSet(VALIDATOR_CONTRACT_ADDR).getMiningValidators();
+
+    bytes memory voteAddress;
+    for (uint i = 0; i < vals.length; i++) {
+      if (vals[i] == _evidence.valAddr) {
+        voteAddress = voteAddrs[i];
+        break;
+      }
+    }
+    bytes memory input;
+    {
+      bytes memory bytesNumA;
+      bytes memory bytesNumB;
+      bytes memory bytesHeaderA;
+      bytes memory bytesHeaderB;
+      TypesToBytes.uintToBytes(32, _evidence.numA, bytesNumA);
+      TypesToBytes.bytes32ToBytes(32, _evidence.headerA, bytesHeaderA);   
+      TypesToBytes.uintToBytes(32, _evidence.numB, bytesNumB);
+      TypesToBytes.bytes32ToBytes(32, _evidence.headerB, bytesHeaderB); 
+
+      input = BytesLib.concat(bytesNumA, bytesHeaderA);
+      input = BytesLib.concat(input, _evidence.sigA);
+      input = BytesLib.concat(input, bytesNumB);
+      input = BytesLib.concat(input, bytesHeaderB);
+      input = BytesLib.concat(input, _evidence.sigB);
+      input = BytesLib.concat(input, voteAddress);
+    }
+
+    bytes memory output;
+    assembly {
+      let len := mload(input)
+      if iszero(call(not(0), 0x64, 0, input, len, output, 0x20)) {  // precompiled contract address 0x64
+        revert(0, 0)
+      }
+    }
+
+    bytes32 _headerA = blockhash(_evidence.numA);
+    bytes32 _headerB = blockhash(_evidence.numB);
+
+    if (_headerA != _evidence.headerA && _headerB != _evidence.headerB) {
+      revert(string(abi.encodePacked("invalid header")));
+    } else if (_headerA == _evidence.headerA && _headerB == _evidence.headerB) {
+      revert(string(abi.encodePacked("invalid evidence")));
+    } else {
+      uint256 amount = (address(SYSTEM_REWARD_ADDR).balance * finalitySlashRewardRatio) / 100;
+      ISystemReward(SYSTEM_REWARD_ADDR).claimRewards(msg.sender, amount);
```

Comment:
Slash contract is not allowed to call the `claimRewards` of system reward contract, because slash contract is not a Operator Account of system reward contract. Please come up with a solution to this.

Reply:
Fixed